#### Software

- [HiCExplorer](https://github.com/deeptools/HiCExplorer/) - set of programs to process, normalize, analyze and visualize Hi-C data, Python, .cool format, conversion utilities. https://hicexplorer.readthedocs.io/en/latest/, https://github.com/deeptools/HiCExplorer/
    - Ramírez, Fidel, Vivek Bhardwaj, Laura Arrigoni, Kin Chung Lam, Björn A. Grüning, José Villaveces, Bianca Habermann, Asifa Akhtar, and Thomas Manke. “[High-Resolution TADs Reveal DNA Sequences Underlying Genome Organization in Flies](https://doi.org/10.1038/s41467-017-02525-w).” Nature Communications 9, no. 1 (December 2018) 

- [FAN-C](https://github.com/vaquerizaslab/fanc) - Python pipeline for Hi-C processing. Input - raw FASTQ (aligned using BWA or Bowtie2, artifact filtering) or pre-aligned BAMs. KR or ICE normalization. Analysis and Visualization (contact distance decay, A/B compartment detection, TAD/loop detection, Average TAD/loop profiles, saddle plots, triangular heatmaps, comparison of two heatmaps).  Automatic or modular. Compatible with .cool and .hic formats. [Tweet1](https://twitter.com/vaquerizas_lab/status/1225011187668209664?s=20), [Tweet2](https://twitter.com/vaquerizasjm/status/1339937903473025027?s=20). [Table 1](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02215-9/tables/1) - detailed comparison of 13 Hi-C processing tools
    - Kruse, Kai, Clemens B. Hug, and Juan M. Vaquerizas. “[FAN-C: A Feature-Rich Framework for the Analysis and Visualisation of Chromosome Conformation Capture Data](https://doi.org/10.1186/s13059-020-02215-9).” Genome Biology 21, no. 1 (December 2020)
    
- [GENOVA](https://github.com/dewitlab/GENOVA) - an R package for the most common Hi-C analyses and visualization. Quality control - cis/trans ratio, checking for translocations; A/B compartments - single and differential compartment signal (H3K4me1 for compartment assignment); TADs - insulation score and directionality index; Genomic annotations - ChIP-seq signal, gene information; Distance decay, and differential analysis; Saddle plot; Aggregate region/peak/TAD/loop analysis, and differential analysis; Aggregated signal (tornado) plots; Intra-inter-TAD interaction differences; PE-SCAn (paired-end Spatial Chromatin Analysis) and C-SCAn enhancer-promoter aggregation. Data for statistical tests can be extracted from the discovery objects. Applied to Hi-C data from Hap1 cells, WT and deltaWAPL (published) and knockdown cohesin-SA1/SA2 conditions (HiC-pro, hg19, HiCCUPS). Input - HiC-pro, Juicer, .cool files. Storage - compressed sparse triangle format and the user-added metadata. All tools return the "discovery object". [GitHub](https://github.com/dewitlab/GENOVA), [Vignette](https://github.com/robinweide/GENOVA/blob/master/vignettes/GENOVA_vignette.pdf)
    - Weide, Robin H. van der, Teun van den Brand, Judith H.I. Haarhuis, Hans Teunissen, Benjamin D. Rowland, and Elzo de Wit. “[Hi-C Analyses with GENOVA: A Case Study with Cohesin Variants](https://doi.org/10.1101/2021.01.22.427620).” Preprint. Genomics, January 24, 2021.
    
- [cooltools](https://github.com/mirnylab/cooltools) - tools to work with .cool files, [Documentation](https://cooltools.readthedocs.io/en/latest/)

- [HiGlass](http://higlass.io/) - visualization server for Google maps-style navigation of Hi-C maps. Overlay genes, epigenomic tracks. "Composable linked views" synchronizing exploration of multiple Hi-C datasets linked by location/zoom. [GitHub](https://github.com/higlass/higlass), [documentation](https://docs.higlass.io/), [links and resources](https://higlass.io/about)
    - Kerpedjiev, Peter, Nezar Abdennur, Fritz Lekschas, Chuck McCallum, Kasper Dinkla, Hendrik Strobelt, Jacob M. Luber, et al. “[HiGlass: Web-Based Visual Exploration and Analysis of Genome Interaction Maps](https://doi.org/10.1186/s13059-018-1486-1).” Genome Biology, (December 2018).

- [Juicebox](https://aidenlab.org/juicebox/)